rules/private/phases/phase_semanticdb.bzl

```diff
@@ -0,0 +1,42 @@
+load("@bazel_skylib//lib:paths.bzl", "paths")
+load("@rules_scala_annex//rules:providers.bzl", _ScalaConfiguration = "ScalaConfiguration")
+
+#
+# PHASE: semanticdb
+#
+# Configures the compiler to output SemanticDB metadata. Note that this phase won't work without the
+# SemanticDB compiler plugin being enabled.
+#
+def phase_semanticdb(ctx, g):
+    scala_configuration = ctx.attr.scala[_ScalaConfiguration]
+
+    if scala_configuration.semanticdb_bundle:
+        return struct(outputs = [], scalacopts = [])
+
+    outputs = []
+    semanticdb_directory = paths.join("_semanticdb/", ctx.label.name)
```

Comment:
Is it possible for the name to contain characters that are illegal to use as file names?

Reply:
From [Bazel's documentation on labels](https://bazel.build/concepts/labels):
> Allowed characters in package names are the lowercase letters `a` through `z`, the uppercase letters `A` through `Z`, the digits `0` through `9`, the characters `! \"#$%&'()*+,-.;<=>?@[]^_`{|}` (yes, there's a space character in there!), and of course forward slash `/` (since it's the directory separator).

I assume legal target names can't include slashes because that'd make the package and name indistinguishable.

I think the only illegal characters in Unix filenames are `/` and the null character, both of which Bazel doesn't allow, so no, it shouldn't be possible. Windows is a different story, but I think it's highly unlikely that special characters other than `-` and `_` will be used anyways, and even if they are, an error should be reported.